### ISSUE TITLE
Exit without error when start training with GPU 

### ISSUE THREAD

**Issue by elechocruz** [PaddlePaddle/PaddleOCR#8472]
请提供下述完整信息以便快速定位问题/Please provide the following information to quickly locate the problem

- 系统环境/System Environment：Windows
- 版本号/Version 10：Paddle：paddlepaddle-gpu 2.4.0.post117  
- 运行指令/Command Code：python tools/train.py -c configs/det/det_mv3_db.yml Global.pretrained_model=./pretrain_models/MobileNetV3_large_x0_5_pretrained
- 完整报错/Complete Error Message：Exit code 3221226505
- GPU: nvidia RTX 2070 Super - CUDA: 11.7 -CUDNN: 8.6

Im trying to train a rec and det model for an ANPR proyect, but when i start te training with the command above using GPU, the process just stop without any error, sometime just throw the Exit Code 3221226505 when train with CPU the command works.
EDIT: After check the windows logs i have an issue on the even viewer that says: 

> Faulting application name: python.exe, version: 3.8.10150.1013, time stamp: 0x608fe38a
> Faulting module name: ucrtbase.dll, version: 10.0.19041.789, time stamp: 0x2bd748bf
> Exception code: 0xc0000409

here is the log when the command is used 

[2022/11/28 12:46:33] ppocr INFO: Architecture :
[2022/11/28 12:46:33] ppocr INFO:     Backbone :
[2022/11/28 12:46:33] ppocr INFO:         last_conv_stride : [1, 2]
[2022/11/28 12:46:33] ppocr INFO:         last_pool_type : avg
[2022/11/28 12:46:33] ppocr INFO:         name : MobileNetV1Enhance
[2022/11/28 12:46:33] ppocr INFO:         scale : 0.5
[2022/11/28 12:46:33] ppocr INFO:     Head :
[2022/11/28 12:46:33] ppocr INFO:         head_list :
[2022/11/28 12:46:33] ppocr INFO:             CTCHead :
[2022/11/28 12:46:33] ppocr INFO:                 Head :
[2022/11/28 12:46:33] ppocr INFO:                     fc_decay : 1e-05
[2022/11/28 12:46:33] ppocr INFO:                 Neck :
[2022/11/28 12:46:33] ppocr INFO:                     depth : 2
[2022/11/28 12:46:33] ppocr INFO:                     dims : 64
[2022/11/28 12:46:33] ppocr INFO:                     hidden_dims : 120
[2022/11/28 12:46:33] ppocr INFO:                     name : svtr
[2022/11/28 12:46:33] ppocr INFO:                     use_guide : True
[2022/11/28 12:46:33] ppocr INFO:             SARHead :
[2022/11/28 12:46:33] ppocr INFO:                 enc_dim : 512
[2022/11/28 12:46:33] ppocr INFO:                 max_text_length : 25
[2022/11/28 12:46:33] ppocr INFO:         name : MultiHead
[2022/11/28 12:46:33] ppocr INFO:     Transform : None
[2022/11/28 12:46:33] ppocr INFO:     algorithm : SVTR
[2022/11/28 12:46:33] ppocr INFO:     model_type : rec
[2022/11/28 12:46:33] ppocr INFO: Eval :
[2022/11/28 12:46:33] ppocr INFO:     dataset :
[2022/11/28 12:46:33] ppocr INFO:         data_dir : ./train_data/data
[2022/11/28 12:46:33] ppocr INFO:         label_file_list : ['./train_data/data/val.txt']
[2022/11/28 12:46:33] ppocr INFO:         name : SimpleDataSet
[2022/11/28 12:46:33] ppocr INFO:         transforms :
[2022/11/28 12:46:33] ppocr INFO:             DecodeImage :
[2022/11/28 12:46:33] ppocr INFO:                 channel_first : False
[2022/11/28 12:46:33] ppocr INFO:                 img_mode : BGR
[2022/11/28 12:46:33] ppocr INFO:             MultiLabelEncode : None
[2022/11/28 12:46:33] ppocr INFO:             RecResizeImg :
[2022/11/28 12:46:33] ppocr INFO:                 image_shape : [3, 48, 320]
[2022/11/28 12:46:33] ppocr INFO:             KeepKeys :
[2022/11/28 12:46:33] ppocr INFO:                 keep_keys : ['image', 'label_ctc', 'label_sar', 'length', 'valid_ratio']
[2022/11/28 12:46:33] ppocr INFO:     loader :
[2022/11/28 12:46:33] ppocr INFO:         batch_size_per_card : 128
[2022/11/28 12:46:33] ppocr INFO:         drop_last : False
[2022/11/28 12:46:33] ppocr INFO:         num_workers : 0
[2022/11/28 12:46:33] ppocr INFO:         shuffle : False
[2022/11/28 12:46:33] ppocr INFO: Global :
[2022/11/28 12:46:33] ppocr INFO:     cal_metric_during_train : True
[2022/11/28 12:46:33] ppocr INFO:     character_dict_path : ppocr/utils/en_dict.txt
[2022/11/28 12:46:33] ppocr INFO:     checkpoints : None
[2022/11/28 12:46:33] ppocr INFO:     debug : False
[2022/11/28 12:46:33] ppocr INFO:     distributed : False
[2022/11/28 12:46:33] ppocr INFO:     epoch_num : 500
[2022/11/28 12:46:33] ppocr INFO:     eval_batch_step : [0, 2000]
[2022/11/28 12:46:33] ppocr INFO:     infer_img : doc/imgs_words/ch/word_1.jpg
[2022/11/28 12:46:33] ppocr INFO:     infer_mode : False
[2022/11/28 12:46:33] ppocr INFO:     log_smooth_window : 20
[2022/11/28 12:46:33] ppocr INFO:     max_text_length : 25
[2022/11/28 12:46:33] ppocr INFO:     pretrained_model : None
[2022/11/28 12:46:33] ppocr INFO:     print_batch_step : 10
[2022/11/28 12:46:33] ppocr INFO:     save_epoch_step : 3
[2022/11/28 12:46:33] ppocr INFO:     save_inference_dir : None
[2022/11/28 12:46:33] ppocr INFO:     save_model_dir : ./output/v3_en_mobile
[2022/11/28 12:46:33] ppocr INFO:     save_res_path : ./output/rec/predicts_ppocrv3_en.txt
[2022/11/28 12:46:33] ppocr INFO:     use_gpu : True
[2022/11/28 12:46:33] ppocr INFO:     use_space_char : True
[2022/11/28 12:46:33] ppocr INFO:     use_visualdl : False
[2022/11/28 12:46:33] ppocr INFO: Loss :
[2022/11/28 12:46:33] ppocr INFO:     loss_config_list :
[2022/11/28 12:46:33] ppocr INFO:         CTCLoss : None
[2022/11/28 12:46:33] ppocr INFO:         SARLoss : None
[2022/11/28 12:46:33] ppocr INFO:     name : MultiLoss
[2022/11/28 12:46:33] ppocr INFO: Metric :
[2022/11/28 12:46:33] ppocr INFO:     ignore_space : False
[2022/11/28 12:46:33] ppocr INFO:     main_indicator : acc
[2022/11/28 12:46:33] ppocr INFO:     name : RecMetric
[2022/11/28 12:46:33] ppocr INFO: Optimizer :
[2022/11/28 12:46:33] ppocr INFO:     beta1 : 0.9
[2022/11/28 12:46:33] ppocr INFO:     beta2 : 0.999
[2022/11/28 12:46:33] ppocr INFO:     lr :
[2022/11/28 12:46:33] ppocr INFO:         learning_rate : 0.001
[2022/11/28 12:46:33] ppocr INFO:         name : Cosine
[2022/11/28 12:46:33] ppocr INFO:         warmup_epoch : 5
[2022/11/28 12:46:33] ppocr INFO:     name : Adam
[2022/11/28 12:46:33] ppocr INFO:     regularizer :
[2022/11/28 12:46:33] ppocr INFO:         factor : 3e-05
[2022/11/28 12:46:33] ppocr INFO:         name : L2
[2022/11/28 12:46:33] ppocr INFO: PostProcess :
[2022/11/28 12:46:33] ppocr INFO:     name : CTCLabelDecode
[2022/11/28 12:46:33] ppocr INFO: Train :
[2022/11/28 12:46:33] ppocr INFO:     dataset :
[2022/11/28 12:46:33] ppocr INFO:         data_dir : ./train_data/data/
[2022/11/28 12:46:33] ppocr INFO:         ext_op_transform_idx : 1
[2022/11/28 12:46:33] ppocr INFO:         label_file_list : ['./train_data/data/train.txt']
[2022/11/28 12:46:33] ppocr INFO:         name : SimpleDataSet
[2022/11/28 12:46:33] ppocr INFO:         transforms :
[2022/11/28 12:46:33] ppocr INFO:             DecodeImage :
[2022/11/28 12:46:33] ppocr INFO:                 channel_first : False
[2022/11/28 12:46:33] ppocr INFO:                 img_mode : BGR
[2022/11/28 12:46:33] ppocr INFO:             RecConAug :
[2022/11/28 12:46:33] ppocr INFO:                 ext_data_num : 2
[2022/11/28 12:46:33] ppocr INFO:                 image_shape : [48, 320, 3]
[2022/11/28 12:46:33] ppocr INFO:                 max_text_length : 25
[2022/11/28 12:46:33] ppocr INFO:                 prob : 0.5
[2022/11/28 12:46:33] ppocr INFO:             RecAug : None
[2022/11/28 12:46:33] ppocr INFO:             MultiLabelEncode : None
[2022/11/28 12:46:33] ppocr INFO:             RecResizeImg :
[2022/11/28 12:46:33] ppocr INFO:                 image_shape : [3, 48, 320]
[2022/11/28 12:46:33] ppocr INFO:             KeepKeys :
[2022/11/28 12:46:33] ppocr INFO:                 keep_keys : ['image', 'label_ctc', 'label_sar', 'length', 'valid_ratio']
[2022/11/28 12:46:33] ppocr INFO:     loader :
[2022/11/28 12:46:33] ppocr INFO:         batch_size_per_card : 128
[2022/11/28 12:46:33] ppocr INFO:         drop_last : True
[2022/11/28 12:46:33] ppocr INFO:         num_workers : 4
[2022/11/28 12:46:33] ppocr INFO:         shuffle : True
[2022/11/28 12:46:33] ppocr INFO: profiler_options : None
[2022/11/28 12:46:33] ppocr INFO: train with paddle 2.4.0 and device Place(gpu:0)
[2022/11/28 12:46:33] ppocr INFO: Initialize indexs of datasets:['./train_data/data/train.txt']
[2022/11/28 12:46:33] ppocr INFO: Initialize indexs of datasets:['./train_data/data/val.txt']
W1128 12:46:33.491047  4804 gpu_resources.cc:61] Please NOTE: device: 0, GPU Compute Capability: 7.5, Driver API Version: 11.8, Runtime API Version: 11.7
W1128 12:46:33.495040  4804 gpu_resources.cc:91] device: 0, cuDNN Version: 8.6.
[2022/11/28 12:46:34] ppocr INFO: train dataloader has 3 iters
[2022/11/28 12:46:34] ppocr INFO: valid dataloader has 1 iters
[2022/11/28 12:46:34] ppocr INFO: train from scratch
[2022/11/28 12:46:34] ppocr INFO: During the training process, after the 0th iteration, an evaluation is run every 2000 iterations

and my yml config file 

Global:
  debug: false
  use_gpu: true
  epoch_num: 500
  log_smooth_window: 20
  print_batch_step: 10
  save_model_dir: ./output/v3_en_mobile
  save_epoch_step: 3
  eval_batch_step: [0, 2000]
  cal_metric_during_train: true
  pretrained_model:
  checkpoints:
  save_inference_dir:
  use_visualdl: false
  infer_img: doc/imgs_words/ch/word_1.jpg
  character_dict_path: ppocr/utils/en_dict.txt
  max_text_length: &max_text_length 25
  infer_mode: false
  use_space_char: true
  distributed: true
  save_res_path: ./output/rec/predicts_ppocrv3_en.txt

Optimizer:
  name: Adam
  beta1: 0.9
  beta2: 0.999
  lr:
    name: Cosine
    learning_rate: 0.001
    warmup_epoch: 5
  regularizer:
    name: L2
    factor: 3.0e-05

Architecture:
  model_type: rec
  algorithm: SVTR
  Transform:
  Backbone:
    name: MobileNetV1Enhance
    scale: 0.5
    last_conv_stride: [1, 2]
    last_pool_type: avg
  Head:
    name: MultiHead
    head_list:
      - CTCHead:
          Neck:
            name: svtr
            dims: 64
            depth: 2
            hidden_dims: 120
            use_guide: True
          Head:
            fc_decay: 0.00001
      - SARHead:
          enc_dim: 512
          max_text_length: *max_text_length

Loss:
  name: MultiLoss
  loss_config_list:
    - CTCLoss:
    - SARLoss:

PostProcess:  
  name: CTCLabelDecode

Metric:
  name: RecMetric
  main_indicator: acc
  ignore_space: False

Train:
  dataset:
    name: SimpleDataSet
    data_dir: ./train_data/data/
    ext_op_transform_idx: 1
    label_file_list:
    - ./train_data/data/train.txt
    transforms:
    - DecodeImage:
        img_mode: BGR
        channel_first: false
    - RecConAug:
        prob: 0.5
        ext_data_num: 2
        image_shape: [48, 320, 3]
        max_text_length: *max_text_length
    - RecAug:
    - MultiLabelEncode:
    - RecResizeImg:
        image_shape: [3, 48, 320]
    - KeepKeys:
        keep_keys:
        - image
        - label_ctc
        - label_sar
        - length
        - valid_ratio
  loader:
    shuffle: true
    batch_size_per_card: 128
    drop_last: true
    num_workers: 4
Eval:
  dataset:
    name: SimpleDataSet
    data_dir: ./train_data/data
    label_file_list:
    - ./train_data/data/val.txt
    transforms:
    - DecodeImage:
        img_mode: BGR
        channel_first: false
    - MultiLabelEncode:
    - RecResizeImg:
        image_shape: [3, 48, 320]
    - KeepKeys:
        keep_keys:
        - image
        - label_ctc
        - label_sar
        - length
        - valid_ratio
  loader:
    shuffle: false
    drop_last: false
    batch_size_per_card: 128
    num_workers: 0

**Comment by jingsongliujing** [comment]
batch_size_per_card : 4

**Comment by elechocruz** [comment]
> batch_size_per_card : 4

omg it works, thank you so much!

**Comment by jingsongliujing** [comment]
This&nbsp;is&nbsp;because&nbsp;the&nbsp;data&nbsp;volume&nbsp;is&nbsp;too&nbsp;large&nbsp;and&nbsp;the&nbsp;memory&nbsp;is&nbsp;not&nbsp;enough for your GPU




---Original---
From: ***@***.***&gt;
Date: Wed, Nov 30, 2022 17:24 PM
To: ***@***.***&gt;;
Cc: ***@***.******@***.***&gt;;
Subject: Re: [PaddlePaddle/PaddleOCR] Exit without error when start trainingwith GPU  (Issue #8472)




 
@jingsongliujing Could you explain why? I'm so confused .....

**Comment by jingsongliujing** [comment]
This is because the data volume is too large and the memory is not enough for your GPU

**Comment by sunzhaoyang** [comment]
@jingsongliujing thanks for your reply. why I got the same situiation when I set `batch_size_per_card: 1` ? too small dataset?
